heron/tools/cli/src/python/execute.py

```diff
@@ -116,30 +121,31 @@ def heron_pex(topology_pex, topology_class_name, args=None):
     cmd = [topology_pex]
     if args is not None:
       cmd.extend(args)
-    Log.debug('$> %s' % ' '.join(cmd))
-    Log.debug('Heron options: %s' % str(heron_env['HERON_OPTIONS']))
+    Log.debug("Invoking class using command: ``%s''", ' '.join(cmd))
+    Log.debug('Heron options: {%s}', str(heron_env['HERON_OPTIONS']))
 
     # invoke the command with subprocess and print error message, if any
-    status = subprocess.call(cmd, env=heron_env)
-    if status != 0:
-      err_str = "Topology failed to be loaded from the given pex, with status: %d. Bailing out..." \
-                % status
-      raise RuntimeError(err_str)
+    proc = subprocess.Popen(cmd, env=heron_env, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    retcode = proc.returncode
+    if retcode != 0:
+      err_str = "Topology class %s failed to be loaded from the given pex" % topology_class_name
+      return response.Response(retcode, err_str, None)
+    else:
+      return response.Response(retcode)
   else:
     try:
       # loading topology from Topology's subclass (no main method)
-
       # to support specifying the name of topology
-      Log.debug("args: %s" % args)
+      Log.debug("args: %s", args)
       if args is not None and isinstance(args, (list, tuple)) and len(args) > 0:
         opts.set_config('cmdline.topology.name', args[0])
-
       os.environ["HERON_OPTIONS"] = opts.get_heron_config()
-      Log.debug("Heron options: %s" % os.environ["HERON_OPTIONS"])
+      Log.debug("Heron options: {%s}", os.environ["HERON_OPTIONS"])
       pex_loader.load_pex(topology_pex)
       topology_class = pex_loader.import_and_get_class(topology_pex, topology_class_name)
       topology_class.write()
-    except Exception:
+      return response.Response(0)
+    except Exception as ex:
       Log.debug(traceback.format_exc())
-      err_str = "Topology failed to be loaded from the given pex. Bailing out..."
-      raise RuntimeError(err_str)
+      err_str = "Topology %s failed to be loaded from the given pex" % topology_class_name
+      return response.Response(100, err_str, str(ex))
```

Comment:
i don't understand the assert part. Where and how should I assert this? Sometimes shelled-out program exits with `> 0 && < 100`, sometimes it could exit with `>100`.